File: Servlet/src/main/java/com/example/servizioripetizioniweb/utils/ErrorCodes.java
package com.example.servizioripetizioniweb.utils;

public enum ErrorCodes {
    ERROR_NOT_AUTHORIZED(403),
    // LOGIN
    ERROR_LOGIN_NOT_LOGGED(450),
    ERROR_LOGIN_WRONG_CREDENTIALS(451),
    ERROR_LOGIN_USER_VALIDATION(452),
    ERROR_LOGIN_PASSWORD_VALIDATION(453),

    // booking
    ERROR_DOCENTE_ACTIVE(459),
    ERROR_CORSO_ACTIVE(460),    // corso is already active, cannot book twice.
    ERROR_ORARIO_ACTIVE(461),   // user has already that hour occupied in another course.

    ERROR_DAY_ERROR(462),
    ERROR_HOUR_ERROR(463),
    ERROR_DOCENTE_ERROR(464),
    ERROR_CORSO_ERROR(465),
    ERROR_PRENOTAZIONE_ERROR(466),
    ERROR_STATO_ERROR(467),
    //generic
    ERROR_UNKNOWN(470);         // generic error. show "try again".



    private int errNum;
    ErrorCodes(int errNum){
        this.errNum = errNum;
    }

    public int getErrNum(){
        return errNum;
    }
}

File: Flutter/lib/utils/error_codes.dart
// ignore_for_file: constant_identifier_names

enum ErrorCodes {
  // LOGIN
  ERROR_LOGIN_NOT_LOGGED(450, 'error_not_logged'),
  ERROR_LOGIN_WRONG_CREDENTIALS(451, 'error_wrong_cred'),
  ERROR_LOGIN_USER_VALIDATION(452, 'error_user_validation'),
  ERROR_LOGIN_PASSWORD_VALIDATION(453, 'error_pw_validation'),

  // booking
  ERROR_DOCENTE_ACTIVE(
      459, 'error_docente_active'), 
  ERROR_CORSO_ACTIVE(
      460, 'error_corso_active'), // corso is already active, cannot book twice.
  ERROR_ORARIO_ACTIVE(461,
      'error_orario_active'), // user has already that hour occupied in another course.

  ERROR_DAY_ERROR(462, 'error_bad_day'),
  ERROR_HOUR_ERROR(463, 'error_bad_hour'),
  ERROR_DOCENTE_ERROR(464, 'error_bad_docente'),
  ERROR_CORSO_ERROR(465, 'error_bad_corso'),
  ERROR_PRENOTAZIONE_ERROR(466, 'error_bad_prenotazione'),
  ERROR_STATO_ERROR(467, 'error_bad_stato'),

  //generic
  ERROR_UNKNOWN(470, 'error_unknown'), // generic error. show "try again".
  ERROR_CONNECTION(499, 'error_connection'),
  ERROR_SERVER(500, 'error_server');

  final int errNum;
  final String message;

  const ErrorCodes(this.errNum, this.message);

  int get errorCode => errNum;
}

final errorMessages = {
  ErrorCodes.ERROR_LOGIN_NOT_LOGGED.errorCode:
      ErrorCodes.ERROR_LOGIN_NOT_LOGGED,
  ErrorCodes.ERROR_LOGIN_WRONG_CREDENTIALS.errorCode:
      ErrorCodes.ERROR_LOGIN_WRONG_CREDENTIALS,
  ErrorCodes.ERROR_LOGIN_USER_VALIDATION.errorCode:
      ErrorCodes.ERROR_LOGIN_USER_VALIDATION,
  ErrorCodes.ERROR_LOGIN_PASSWORD_VALIDATION.errorCode:
      ErrorCodes.ERROR_LOGIN_PASSWORD_VALIDATION,
  ErrorCodes.ERROR_DOCENTE_ACTIVE.errorCode: ErrorCodes.ERROR_DOCENTE_ACTIVE,
  ErrorCodes.ERROR_CORSO_ACTIVE.errorCode: ErrorCodes.ERROR_CORSO_ACTIVE,
  ErrorCodes.ERROR_ORARIO_ACTIVE.errorCode: ErrorCodes.ERROR_ORARIO_ACTIVE,
  ErrorCodes.ERROR_DAY_ERROR.errorCode: ErrorCodes.ERROR_DAY_ERROR,
  ErrorCodes.ERROR_HOUR_ERROR.errorCode: ErrorCodes.ERROR_HOUR_ERROR,
  ErrorCodes.ERROR_DOCENTE_ERROR.errorCode: ErrorCodes.ERROR_DOCENTE_ERROR,
  ErrorCodes.ERROR_CORSO_ERROR.errorCode: ErrorCodes.ERROR_CORSO_ERROR,
  ErrorCodes.ERROR_PRENOTAZIONE_ERROR.errorCode:
      ErrorCodes.ERROR_PRENOTAZIONE_ERROR,
  ErrorCodes.ERROR_STATO_ERROR.errorCode: ErrorCodes.ERROR_STATO_ERROR,
  ErrorCodes.ERROR_UNKNOWN.errorCode: ErrorCodes.ERROR_UNKNOWN,
  ErrorCodes.ERROR_CONNECTION.errorCode: ErrorCodes.ERROR_CONNECTION,
  ErrorCodes.ERROR_SERVER.errorCode: ErrorCodes.ERROR_SERVER,
};
